src/js/tabs/power.js

```diff
@@ -281,6 +297,47 @@ TABS.power.initialize = function (callback) {
 
         }
 
+        //calibration manager
+        OSD.GUI.calibrationManager = new jBox('Modal', {
+            width: 360,
+            height: 180,
+            closeButton: 'title',
+            animation: false,
+            attach: $('#calibrationmanager'),
+            title: 'Calibration Manager',
+            content: $('#calibrationmanagercontent'),
+        });
+
+        $('input[name="vbatcalibration"]').val(0);
+        $('input[name="amperagecalibration"]').val(0);
+
+        $('a.savecalibration').click(function(){
+            if (BATTERY_CONFIG.voltageMeterSource == 1) {
+                var vbatcalibration = parseFloat($('input[name="vbatcalibration"]').val());
+                if (vbatcalibration != 0) {
+                    var vbatnewscale = Math.round(VOLTAGE_METER_CONFIGS[0].vbatscale * (vbatcalibration / VOLTAGE_METERS[0].voltage));
+                    if (vbatnewscale >= 10 && vbatnewscale <= 255) {
+                        VOLTAGE_METER_CONFIGS[0].vbatscale = vbatnewscale;
+                    }
+                }
+            }
+            var ampsource = BATTERY_CONFIG.currentMeterSource;
+            if (ampsource == 1 || ampsource == 2) {
+                var amperagecalibration = parseFloat($('input[name="amperagecalibration"]').val());
+                var amperageoffset = CURRENT_METER_CONFIGS[ampsource - 1].offset / 1000;
+                if (amperagecalibration != 0) {
+                    if (CURRENT_METERS[ampsource - 1].amperage != amperageoffset && amperagecalibration != amperageoffset) {
+                        var amperagenewscale = Math.round(CURRENT_METER_CONFIGS[ampsource - 1].scale * 
+                            ((CURRENT_METERS[ampsource - 1].amperage -  amperageoffset) / (amperagecalibration - amperageoffset)));
+                        if (amperagenewscale > -16000 && amperagenewscale < 16000 && amperagenewscale != 0) {
+                            CURRENT_METER_CONFIGS[ampsource - 1].scale = amperagenewscale;
+                        }
+                    }
+                }
+            }
+            save_power_config();
```

Comment:
This should not be saving the settings - the user might have changed other things in the 'Power' tab that they do not want to keep. Better don't save, but add 'Remember to save your settings after calibrating' to the note in the dialog.

Reply:
It only saves scales which are calculated, and no other settings that are changed in power tab are saved. When sources for voltage or current are changed the calibration manager is disabled until the sources are saved.

Reply:
Looking at the code, this is simply not true. It will save **all** settings on the tab, including warning voltages and capacity. The real problem is that it does not give the user any indication that it will do so, potentially overwriting settings that they did not want to overwrite, and it does not allow user to verify that the 'calibrated' settings are actually what they want (they might be obviously out of range, especially for the amperage calibration, where calibrating at idle current without motors running will be extremely prone to be way off).
So, in order to have a good UX, the right thing to do is to present the calibration result to the user as suggestions, and allow them to then decide if they want to save them or not.

Reply:
I understand what would be the problem in that case, but I am looking at the code and testing it just now and it is not saving all settings. save_power_config() indeed does save all settings but if they are not modified by parsing the values they stay the same, so only calibration scales get set in CURRENT_METER_CONFIGS and VOLTAGE_METER_CONFIGS. Other settings on tab only get saved when actual save button is pressed and values are parsed.

Reply:
Do you think the confirmation dialog of new calculated scales showing the new voltage/current with option to apply or discard them would be a good idea?

Reply:
> I understand what would be the problem in that case, but I am looking at the code and testing it just now and it is not saving all settings. save_power_config() indeed does save all settings but if they are not modified by parsing the values they stay the same,

This is even worse, as this results in buggy behaviour: After closing the calibration dialog, the values displayed to the user in the battery tab will be a mix of saved and unsaved values.

> Do you think the confirmation dialog of new calculated scales showing the new voltage/current with option to apply or discard them would be a good idea?

I think it will be sufficient to allow the user to use the battery tab to confirm the new scale values before saving. Maybe add text to the dialog like 'after calibration, check that the new voltage / amperage scale values are within the expected range, and then hit [Save] to persist them.'

Reply:
After closing they get refreshed, meaning they are all saved values. Anyway, I will make a confirmation dialog.